### PR TITLE
sb3: reinstate couchbase samples

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/issues/19782

Can be used as base to fix couchbase support.

Reverts jhipster/generator-jhipster#20564

### CODE REVIEW

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -375,6 +375,8 @@ if (SPRING_BOOT_VERSION.indexOf('M') > -1 || SPRING_BOOT_VERSION.indexOf('RC') >
 <%_ if (databaseTypeCouchbase) { _%>
     testImplementation "org.testcontainers:couchbase"
     implementation "com.couchbase.client:java-client"
+    // TODO drop once spring-boot 3.0.5 is released
+    implementation "org.springframework.data:spring-data-couchbase:<%- javaDependencies['spring-data-couchbase'] %>"
```
Comment (mshima) [inline generators/server/templates/build.gradle.ejs:379]:
```suggestion
```

File: generators/server/templates/pom.xml [hunk 2]
```diff
@@ -50,6 +50,7 @@
         <openapi-generator-maven-plugin.version>6.4.0</openapi-generator-maven-plugin.version>
         <properties-maven-plugin.version>1.1.0</properties-maven-plugin.version>
         <sonar-maven-plugin.version>3.9.1.2184</sonar-maven-plugin.version>
+        <spring-data-couchbase.version>5.0.4</spring-data-couchbase.version>
```
Comment (mshima) [inline generators/server/templates/pom.xml:53]:
```suggestion
```

File: generators/server/templates/pom.xml [hunk 3]
```diff
@@ -124,6 +125,12 @@
               <artifactId>mapstruct</artifactId>
               <version>${mapstruct.version}</version>
             </dependency>
+            <!-- TODO drop once spring-boot 3.0.5 is released -->
+            <dependency>
+                <groupId>org.springframework.data</groupId>
+                <artifactId>spring-data-couchbase</artifactId>
+                <version>${spring-data-couchbase.version}</version>
+            </dependency>
```
Comment (mshima) [inline generators/server/templates/pom.xml:133]:
```suggestion
```

File: generators/server/templates/pom.xml.ejs [hunk 4]
```diff
@@ -399,6 +399,12 @@
             <groupId>com.couchbase.client</groupId>
             <artifactId>java-client</artifactId>
         </dependency>
+        <!-- TODO drop once spring-boot 3.0.5 is released -->
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-couchbase</artifactId>
+            <version><%- javaDependencies['spring-data-couchbase'] %></version>
+        </dependency>
```
Comment (mshima) [inline generators/server/templates/pom.xml.ejs:407]:
```suggestion
```

**Comment by DanielFran** [comment]
@tchlyah Any help is much appreciated.

**Comment by mraible** [comment]
Merged in the `main` branch to see if things work now.

**Comment by mshima** [comment]
Looks like there is no interest in Couchbase.
Open for too long, closing.

**Pull request closed without merging**